File: Framework.D-2015/Framework.D-2015/Serializadores/Serializador.vb
Imports System.Xml.Serialization
Imports System.IO
Imports Framework.D_2015.Funciones
Namespace Serializadores

    Public Class SerializadorXml
        Implements ISerializador


        Public Function Serializar(unObjeto As Object) As String Implements ISerializador.Serializar

            Dim xmlSerializer As New XmlSerializer(unObjeto.GetType())

            Dim ms As New System.IO.MemoryStream

            xmlSerializer.Serialize(ms, unObjeto)

            Return Cadenas.StreamACadena(ms)

        End Function

        Public Function Deserealizar(Of T As New)(unObjetoSerializadoEnXml As String) As T Implements ISerializador.Deserealizar

            Dim xmlSerializer As New XmlSerializer(GetType(T))

            Dim resultado = xmlSerializer.Deserialize(New MemoryStream(Cadenas.CadenaABytes(unObjetoSerializadoEnXml)))

            Return CType(resultado, T)

        End Function

    End Class
End Namespace

File: Framework.D-2015/Framework.D-2015/Serializadores/Serializador.cs
using System.IO;
using System.Xml;
using System.Xml.Serialization;

namespace Framework.D_2015.Serializadores
{
    public class SerializadorXml : ISerializador
    {
        public string Serializar(object unObjeto)
        {
            var xmlSerializer = new XmlSerializer(unObjeto.GetType());
            var ms = new MemoryStream();
            xmlSerializer.Serialize(ms, unObjeto);
            return Cadenas.StreamACadena(ms);
        }

        public T Deserealizar<T>(string unObjetoSerializadoEnXml) where T : new()
        {
            var xmlSerializer = new XmlSerializer(typeof(T));
            var resultado = xmlSerializer.Deserialize(new MemoryStream(Cadenas.CadenaABytes(unObjetoSerializadoEnXml)));
            return (T)resultado;
        }
    }
}